media/js/wiki-samples.js

```diff
@@ -0,0 +1,88 @@
+(function(win, doc, $) {
+
+    var sites = ['jsfiddle', 'codepen'];
+
+    // using id to get sample code section since macros discard other attributes
+    $('.sample-code-frame').before(function() {
+        var section = $(this).attr('id').substring("frame_".length);
+        return createSampleButtons(section, sites);
+    });
+    $('.sample-code-table').before(function(){
+        var section = $(this).find('iframe').attr('id').substring("frame_".length);
+        return createSampleButtons(section, sites);
+    });
+
+    $('#wikiArticle').on('click', 'button.open-in-host', function(){
+        var $button = $(this);
+        var section = $button.attr('data-section');
+        var sampleCodeHost = $button.attr('data-host');
+        var sampleUrl = win.location.href.split('#')[0] + '?section=' + section + '&raw=1';
+
+        // track the click and sample code host
+        mdn.analytics.trackEvent({
+            category: 'Samples',
+            action: 'open-' + sampleCodeHost,
+            label: section
+        });
+
+        // disable the button, till we open the fiddle
+        $button.attr('disabled', 'disabled');
+        $.get(sampleUrl).then(function(sample) {
+            var $sample = $('<div />').append(sample);
+            var htmlCode = $sample.find('.brush\\:.html, .brush\\:.html\\;').text();
+            var cssCode = $sample.find('.brush\\:.css, .brush\\:.css\\;').text();
+            var jsCode = $sample.find('.brush\\:.js, .brush\\:.js\\;').text();
+            var title = $sample.find('h2[name=' + section + ']').text();
+            openSample(sampleCodeHost, title, htmlCode, cssCode, jsCode);
+
+            $button.removeAttr('disabled');
+        });
+    });
+
+    function openJSFiddle(title, htmlCode, cssCode, jsCode) {
+       var $form = $('<form method="post" target="_blank" action="https://jsfiddle.net/api/mdn/" class="hidden">'
+            + '<input type="hidden" name="html" />'
```

Comment:
JSHint suggests the "+" for each of these multi-line declarations should go on the preceding line.